src/components/Button.tsx

```diff
@@ -189,52 +191,60 @@ const DefaultButton = Button.extend`
 	}
 `;
 
-export default withTheme(
-	asRendition(
-		({
-			outline,
-			underline,
-			plaintext,
-			children,
-			iconElement,
-			...props
-		}: ThemedButtonProps) => {
-			if (plaintext) {
-				return (
-					<Plaintext {...props}>
-						{iconElement && <ButtonIcon>{iconElement}</ButtonIcon>}
-						{children}
-					</Plaintext>
-				);
-			} else if (outline) {
-				return (
-					<Outline {...props}>
-						{iconElement && <ButtonIcon>{iconElement}</ButtonIcon>}
-						{children}
-					</Outline>
-				);
-			} else if (underline) {
-				return (
-					<Underline {...props}>
-						{iconElement && <ButtonIcon>{iconElement}</ButtonIcon>}
-						{children}
-					</Underline>
-				);
-			} else if (!getColoringType(props) && !props.color && !props.bg) {
-				return (
-					<DefaultButton {...props}>
-						{iconElement && <ButtonIcon>{iconElement}</ButtonIcon>}
-						{children}
-					</DefaultButton>
-				);
-			} else {
-				return (
-					<Button {...props}>
-						{iconElement && <ButtonIcon>{iconElement}</ButtonIcon>}
-						{children}
-					</Button>
-				);
+const getStyledButton = ({
+	outline,
+	underline,
+	plaintext,
+	...props
+}: ThemedButtonProps) => {
+	if (plaintext) {
+		return Plaintext;
+	}
+	if (outline) {
+		return Outline;
+	}
+	if (underline) {
+		return Underline;
+	}
+	if (!getColoringType(props) && !props.color && !props.bg) {
+		return DefaultButton;
+	}
+	return ButtonBase;
+};
+
+const ButtonFactory = <T extends keyof JSX.IntrinsicElements>(tag?: T) =>
+	withTheme(
+		asRendition((props: ThemedButtonProps) => {
+			const {
```

Comment:
NIT: It's maybe a bit cleaner if the function argument to `asRendition` is extracted and the ButtonFactory reads as 
`const ButtonFactory = <T extends keyof JSX.IntrinsicElements>(tag?: T) => withTheme(asRendition(Button))`.